demos/common/cpp/models/src/detection_model_centernet.cpp

```diff
@@ -0,0 +1,307 @@
+}
+
+void ModelCenterNet::prepareInputsOutputs(InferenceEngine::CNNNetwork& cnnNetwork) {
+    // --------------------------- Configure input & output -------------------------------------------------
+    // --------------------------- Prepare input blobs ------------------------------------------------------
+    slog::info << "Checking that the inputs are as the demo expects" << slog::endl;
+    InferenceEngine::InputsDataMap inputInfo(cnnNetwork.getInputsInfo());
+    if (inputInfo.size() != 1) {
+        throw std::logic_error("This demo accepts networks that have only one input");
+    }
+
+    InferenceEngine::InputInfo::Ptr& input = inputInfo.begin()->second;
+    const InferenceEngine::TensorDesc& inputDesc = input->getTensorDesc();
+    input->setPrecision(InferenceEngine::Precision::U8);
+
+    if (inputDesc.getDims()[1] != 3) {
+        throw std::logic_error("Expected 3-channel input");
+    }
+
+    if (useAutoResize) {
+        input->getPreProcess().setResizeAlgorithm(InferenceEngine::ResizeAlgorithm::RESIZE_BILINEAR);
+        input->getInputData()->setLayout(InferenceEngine::Layout::NHWC);
+    }
+    else {
+        input->getInputData()->setLayout(InferenceEngine::Layout::NCHW);
+    }
+
+    // --------------------------- Reading image input parameters -------------------------------------------
+    std::string imageInputName = inputInfo.begin()->first;
+    inputsNames.push_back(imageInputName);
+    netInputHeight = getTensorHeight(inputDesc);
+    netInputWidth = getTensorWidth(inputDesc);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+
+    InferenceEngine::OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    if (outputInfo.size() != 3) {
+        throw std::logic_error("This demo expect networks that have 3 outputs blobs");
+    }
+
+    const InferenceEngine::TensorDesc& outputDesc = outputInfo.begin()->second->getTensorDesc();
+    for (auto& output : outputInfo) {
+        output.second->setPrecision(InferenceEngine::Precision::FP32);
+        output.second->setLayout(InferenceEngine::Layout::NCHW);
+        outputsNames.push_back(output.first);
+    }
+
+}
+
+cv::Point2f getDir(const cv::Point2f& srcPoint, float rotRadius) {
+    float sn = sinf(rotRadius);
+    float cs = cosf(rotRadius);
+
+    cv::Point2f srcResult(0.0f, 0.0f);
+    srcResult.x = srcPoint.x * cs - srcPoint.y * sn;
+    srcResult.y = srcPoint.x * sn + srcPoint.y * cs;
+
+    return srcResult;
+}
+
+cv::Point2f get3rdPoint(const cv::Point2f& a, const cv::Point2f& b) {
+    cv::Point2f direct = a - b;
+    return b + cv::Point2f(-direct.y, direct.x);
+}
+
+cv::Mat getAffineTransform(float centerX, float centerY, int scale, float rot, size_t outputWidth, size_t outputHeight, bool inv = false) {
+    int srcW = scale;
```

Comment:
You can rename function arg instead of copying the value.

Reply:
ok